ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt

```diff
@@ -79,7 +80,13 @@ public class JavaHttpEngine(override val config: JavaHttpConfig) : HttpClientEng
 
                 data.getCapabilityOrNull(HttpTimeout)?.let { timeoutAttribute ->
                     timeoutAttribute.connectTimeoutMillis?.let {
-                        connectTimeout(Duration.ofMillis(it))
+                        try {
+                            // Check that timeout end date as the number of milliseconds can fit Long type
+                            Instant.now().plus(it, ChronoUnit.MILLIS).toEpochMilli()
```

Comment:
What's the benefit of checking it this way instead of just comparing with `INFINITE_TIMEOUT_MS`? Is it realistic that users will set timeout duration higher then `Long.MaxValue - now()`?